pkg/kdk/config.go

```diff
@@ -0,0 +1,123 @@
+// Copyright © 2018 Cisco Systems, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package kdk
+
+import (
+	"context"
+	"os/user"
+	"path/filepath"
+	"strings"
+
+	"github.com/docker/docker/api/types/container"
+	"github.com/docker/docker/client"
+	"github.com/mitchellh/go-homedir"
+)
+
+// Struct of all configs to be saved directly as ~/.kdk/<NAME>/config.yaml
+type KdkEnvConfig struct {
+	DockerClient *client.Client
+	Ctx          context.Context
+	ConfigFile   configFile
+}
+
+type configFile struct {
+	AppConfig       AppConfig
+	ContainerConfig *container.Config     `json:",omitempty"`
+	HostConfig      *container.HostConfig `json:",omitempty"`
+}
+
+type AppConfig struct {
+	Name            string
+	Port            string
+	ImageRepository string
+	ImageTag        string
+	DotfilesRepo    string
+	Shell           string
+	Debug           bool
+}
+
+// Struct of configs from the docker lib, to be saved directly as ~/.kdk/config.yaml
+type kdkConfig struct {
+	ContainerConfig container.Config     `json:",omitempty"`
+	HostConfig      container.HostConfig `json:",omitempty"`
+}
+
+// create docker client and context for easy reuse
+func (c *KdkEnvConfig) Init() {
+	c.Ctx = context.Background()
+	dockerClient, err := client.NewEnvClient()
+	if err != nil {
+		panic(err)
+	}
+	c.DockerClient = dockerClient
+}
+
+// current username
+func (c *KdkEnvConfig) User() (out string) {
+	currentUser, err := user.Current()
```

Comment:
`c` doesn't appear to be used in this method. Perhaps this should be a `SetUser()` method and another `User()` could be made to just read.  That way testing as other users could be more easily done. 

(may)